legion/engine/llri/detail/flags.hpp

```diff
@@ -0,0 +1,87 @@
+/**
+ * @file flags.hpp
+ * @copyright 2021-2021 Leon Brands. All rights served.
+ * @license: https://github.com/Legion-Engine/Legion-LLRI/blob/main/LICENSE
+ */
+
+#pragma once
+#include <type_traits>
+
+namespace LLRI_NAMESPACE
+{
+    /**
+     * @brief wrapper structure for flags of type E, used to distinguish between x_flag_bits and x_flags.
+    */
+    template<typename E>
+    struct flags
+    {
+        using U = std::underlying_type_t<E>;
```

Comment:
since U is a public alias, maybe use a more descriptive name?

Reply:
sure! just wasn't sure what to name it

Reply:
maybe just `underlying_t`

Reply:
that works